### PR TITLE
[CALCITE-5922] The SQL generated for the POSITION function(with 3 input arguments) by the SparkSqlDialect is not recognized by Spark SQL

### PR DESCRIPTION
**Pull request by macroguo-ghy**

### CODE REVIEW

**Review by JiajunBernoulli — APPROVED** [review]

**Comment by macroguo-ghy** [comment]
Hi @JiajunBernoulli , I have made a change to the git comment and JIRA summary without any other modifications. Would you like to review it again and then merge this PR?

File: core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java [hunk 1]
```diff
@@ -147,9 +147,34 @@ public SparkSqlDialect(SqlDialect.Context context) {
       case TRIM:
         unparseHiveTrim(writer, call, leftPrec, rightPrec);
         break;
+      case POSITION:
```
Comment (NobiGo) [inline core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java:150]: I noticed `POSITION('a', 'abc')` and `POSITION('a' IN 'abc')` all works in Spark. So we can use 
` case POSITION:
    SqlUtil.unparseFunctionSyntax(SqlStdOperatorTable.POSITION, writer, call, false);`. WDYT?

**Review by NobiGo — CHANGES_REQUESTED** [review]

**Review by NobiGo — APPROVED** [review]
LGTM

**Comment by JiajunBernoulli** [comment]
@macroguo-ghy  Please squash commits, I will merge it tomorrow.

**Comment by macroguo-ghy** [comment]
@JiajunBernoulli  Done.

**Pull request merged**